README.md

```diff
@@ -1,127 +1,124 @@
+
+Aside from the tuple issue, iterating over a dictionary looks the same as iterating over an array:
 
 ```swift
 for (key, value) in airports {
-    print("The abbreviation for the \(value) is \(key)")
+    print("The abbreviation for \(key) is \(value)")
 }
-
-// prints "The abbreviation for the Dublin is DUB"
-// prints "The abbreviation for the Toronto Pearson is YYZ"
-// prints "The abbreviation for the John F. Kennedy International is JFK"
+// prints The abbreviation for Dublin is DUB
+// prints The abbreviation for Toronto Pearson is YYZ
+// prints The abbreviation for John F. Kennedy International Airport is JFK
 ```
 
-* Step through exactly what's going on here. At this point, they should know what a tuple is... but briefly cover it again. Either way, I also want them to know they could name those whatever they want.. however it fits their problem at hand:
+Since iterating over the `airports` dictionary returns a tuple containing both the key and value, you can _unpack_ that tuple in the first line of the for loop. In the example above, the dictionary's key is bound to `key` in every iteration, and the value is bound to `value`. Within the body of the for loop, `key` and `value` can be used like any other constant.
+
+Pretty easy, right?
+
+You don't have to use `key` and `value` to refer to the items returned in each iteration. You can use any name you want. `airportCode` and `airportName` might be more clear, in this case, so this works, too:
 
 ```swift
 for (airportCode, airportName) in airports {
-    print("The abbreviation for the \(airportName) is \(airportCode)")
+    print("The abbreviation for \(airportName) is \(airportCode)")
 }
-
-// prints "The abbreviation for the Dublin is DUB"
-// prints "The abbreviation for the Toronto Pearson is YYZ"
-// prints "The abbreviation for the John F. Kennedy International is JFK"
+// prints The abbreviation for Dublin is DUB
+// prints The abbreviation for Toronto Pearson is YYZ
+// prints The abbreviation for John F. Kennedy International Airport is JFK
 ```
 
-* No difference from the one above.
-* Go over the definition of `for in` again with the student here. They will have seen it only once before at this point in relation to Arrays.
+Iterating over dictionaries isn't much different than iterating over arrays. Did you notice something surprising about the output, though? The original dictionary you defined had a different order than the output. The original dictionary had keys in the order "YYZ", "DUB", and "JFK", but the dictionary output them in the order "DUB", "YYZ", and "JFK". This is because dictionaries are _unordered_. Unlike arrays, the order in which you put items is not necessarily the order you get them out when iterating. Sometimes you will get output in the same order, but this is not guaranteed.
 
-* I want them to be able to work with an array as a value for keys like such:
+## Much Ado About Dictionaries
```

Comment:
I had seen "Much Ado About Nothing" here in Central Park... it was Shakespeare In The park. It was one of the coolest, funniest things I've ever experienced.